security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -150,6 +150,9 @@ func (c *CitadelClient) getTLSDialOption() (grpc.DialOption, error) {
 		RootCAs: certPool,
 	}
 
+	// strip the port from the address
+	parts := strings.Split(c.opts.CAEndpoint, ":")
```

Comment:
net.SplitHostPort may be more appropriate - this could fail on an ipv6 address for example.
Setting this when it's an IP may fail in general actually? not certain

Reply:
Right, i will fix the xds client as well